### PR TITLE
Fix template expressions for dotnet

### PR DESCRIPTION
**Pull request by viveklak**
Fix for #6265

### CODE REVIEW

File: pkg/codegen/dotnet/gen_program_expressions.go [hunk 1]
```diff
@@ -519,16 +519,21 @@ func (g *generator) GenSplatExpression(w io.Writer, expr *model.SplatExpression)
...
 	multiLine := false
 	expressions := false
+	hasOutput := false
 	for _, expr := range expr.Parts {
 		if lit, ok := expr.(*model.LiteralValueExpression); ok && lit.Type() == model.StringType {
 			if strings.Contains(lit.Value.AsString(), "\n") {
 				multiLine = true
 			}
 		} else {
+			hasOutput = true
```
Comment (viveklak) [inline pkg/codegen/dotnet/gen_program_expressions.go:529]: Not sure if we need to be more selective around when to wrap in outputs here?

Comment (mikhailshilkov) [reply pkg/codegen/dotnet/gen_program_expressions.go:529]: There is a test failure - the new code is `Source = new FileAsset(Output.Format($"{siteDir}/{range.Value}")),` which won't work. It needs to be something like `Source = Output.Format($"{siteDir}/{range.Value}").Apply(v => (AssetOrArchive)new FileAsset(v)),`.

I guess we need to change the return type of this expression to an Output, not just add the `Output.Format` mechanically.

However, your code may be "less" broken - so maybe it's okay as the first step? How do other SDKs solve that?

Comment (mikhailshilkov) [reply pkg/codegen/dotnet/gen_program_expressions.go:529]: On the second look, those tests aren't outputs, so they shouldn't actually change - they seem fine as they were before the change. So yes, I think we should also be more selective.

Comment (viveklak) [reply pkg/codegen/dotnet/gen_program_expressions.go:529]: Yes I was looking at the test failure as well. Will try to modify the expression as you suggested and am making the `Output.Format` more selective.

Comment (pgavlin) [reply pkg/codegen/dotnet/gen_program_expressions.go:529]: In other code generators we drive this off of pattern-matching on applies. That's probably what we should do here as well. @viveklak do you want to take that on? If not, I can take it.

Comment (viveklak) [reply pkg/codegen/dotnet/gen_program_expressions.go:529]: @pgavlin yes sorry I have obviously lost track of this PR :(. I'd appreciate it if you could take over!

**Pull request closed without merging**